jep/0000/README.adoc

```diff
@@ -0,0 +1,366 @@
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>plugin</artifactId>
+        <version>4.56</version>
+        <relativePath />
+    </parent>
+    <artifactId>karl</artifactId>
+    <version>${revision}${changelist}</version>
+    <packaging>hpi</packaging>
+    <name>Karl Plugin</name>
+    <url>
+        <!--Something like https://wiki.jenkins-ci.org/display/JENKINS/Karl+Plugin-->
+    </url>
+    <description>A useful description of the Karl plugin.</description>
+    <licenses>
+        <license>
+            <name>MIT</name>
+            <url>http://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
+
+    <properties>
+        <revision>3.1.4</revision>
+        <changelist>-SNAPSHOT</changelist>
+        <java.level>8</java.level>
+        <jenkins.version>2.138.4</jenkins.version>
+    </properties>
+
+----
+
+No additional changes need to be made to `pom.xml` by Karl, he's good to go.
+
+==== Autorelease.yaml verification in `buildPlugin`
+
+The commonly used library link:https://github.com/jenkins-infra/pipeline-library/blob/master/vars/buildPlugin.groovy[`buildPlugin`] will need to be modified to check for the presence of, and validate, `Autorelease.yaml`. This 
+markerfile will be the mechanism that tells the trusted CI server that this plugin should be automatically 
+released. 
+
+Validation must include, but will not be limited to, the following:
+
+1. `pluginName` field must match the repository name
+
+2. `versionNumberFields` must match the `revision` property in `pom.xml`. E.g., if `versionNumberFields` is 
+set to `3` (three fields), and the `revision` in `pom.xml` is set to `3.1.4.5` (four fields), validation 
+will fail.
+
+3. Validation that no extra fields are present in the file. Comments are allowed, but any unexpected 
+lines will cause validation to fail.
+
+If validation of `Autorelease.yaml` fails for any reason, the build is not performed, nothing gets deployed, 
+and GitHub is notified of the failure.
+
+==== Deployment of a successful release
+
+When Karl merges a commit into the `master` branch, that merge commit is link:https://github.com/jenkinsci/jep/blob/master/jep/305/README.adoc#basic-usage[commit number] 150, and has the 
+SHA `1a2b3c4`. The following takes place:
+
+* A webhook is sent from GitHub, to the webhook receiver. That hook is forwarded  to 
+the trusted CI server.
+* `Autorelease.yaml` is validated.
+* Once validation passes, a build is performed. If the build passes all its tests, a release 
+is generated. In our example, that release number would be `3.1.4-150-1a2b3c4`
+* The built plugin gets deployed to Artifactory
+* link:https://github.com/jenkins-infra/update-center2/blob/master/README.md[The UC generator] makes
```

Comment:
I firmly believe NOT creating a tag is a bad idea.

Reply:
Can you write up your objections in a form that can be incorporated into **Reasoning**?

Reply:
@jtnord You can provide further feedback after this PR is merged.

Reply:
I've moved the `WARNING` section and changed the link so that it points [here](https://github.com/jenkinsci/jep/pull/244/files#r302487016). We'll come back to this.

Reply:
The anchor on that URL does not resolve for me, FWIW.

Reply:
This is pretty odd. I found what's wrong but I've got no idea how it got that way. URL needs to point to `discussion#anchor` rather than `files#anchor`.

Reply:
developers are used to being able to see and work with tags for releases.  it makes any tooling that needs to checkout a release work without changes (if a plugin is auto released or not should not matter, and it allows you to not worry if something was auto-released or not.  TL;DR this is not the way that maven users expect things to work)